### PR TITLE
r/iam_oidc_provider - add tagging support + validations + sweeper

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #17594

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIAMOpenIDConnectProvider_'
--- PASS: TestAccAWSIAMOpenIDConnectProvider_disappears (27.34s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_basic (56.82s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_tags (78.49s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSIAMOpenIDConnectProvider_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSIAMOpenIDConnectProvider_ -timeout 120m
=== RUN   TestAccAWSIAMOpenIDConnectProvider_basic
=== PAUSE TestAccAWSIAMOpenIDConnectProvider_basic
=== RUN   TestAccAWSIAMOpenIDConnectProvider_tags
=== PAUSE TestAccAWSIAMOpenIDConnectProvider_tags
=== RUN   TestAccAWSIAMOpenIDConnectProvider_disappears
=== PAUSE TestAccAWSIAMOpenIDConnectProvider_disappears
=== CONT  TestAccAWSIAMOpenIDConnectProvider_basic
=== CONT  TestAccAWSIAMOpenIDConnectProvider_disappears
=== CONT  TestAccAWSIAMOpenIDConnectProvider_tags
--- PASS: TestAccAWSIAMOpenIDConnectProvider_disappears (11.47s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_basic (23.76s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_tags (31.17s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	31.281s
```

**Comment by ewbankkit** [comment]
Ran sweeper:

```console
$ TEST=./aws SWEEP=us-west-2 SWEEPARGS=-sweep-run=aws_iam_openid_connect_provider make sweep
WARNING: This will destroy infrastructure. Use only in development accounts.
go test ./aws -v -sweep=us-west-2 -sweep-run=aws_iam_openid_connect_provider -timeout 60m
2021/03/06 16:48:56 [DEBUG] Running Sweepers for region (us-west-2):
2021/03/06 16:48:56 [DEBUG] Running Sweeper (aws_iam_openid_connect_provider) in region (us-west-2)
2021/03/06 16:48:56 [INFO] AWS Auth provider used: "EnvProvider"
2021/03/06 16:48:56 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2021/03/06 16:48:56 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2021/03/06 16:48:57 Sweeper Tests ran successfully:
	- aws_iam_openid_connect_provider
ok  	github.com/terraform-providers/terraform-provider-aws/aws	0.851s
```

File: aws/resource_aws_iam_openid_connect_provider_test.go [hunk 1]
```diff
@@ -2,16 +2,63 @@ package aws
...
+func testSweepIamOpenIDConnectProvider(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).iamconn
+
+	var sweeperErrs *multierror.Error
+
+	out, err := conn.ListOpenIDConnectProviders(&iam.ListOpenIDConnectProvidersInput{})
```
Comment (bflad) [inline aws/resource_aws_iam_openid_connect_provider_test.go:32]: Verified this does not have an available paginator or token to manually do so.

File: aws/resource_aws_iam_openid_connect_provider_test.go [hunk 2]
```diff
@@ -50,6 +99,51 @@ func TestAccAWSIAMOpenIDConnectProvider_basic(t *testing.T) {
...
+}
+
+func TestAccAWSIAMOpenIDConnectProvider_tags(t *testing.T) {
+	rString := acctest.RandString(5)
+	resourceName := "aws_iam_openid_connect_provider.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSInstanceProfileDestroy,
```
Comment (bflad) [inline aws/resource_aws_iam_openid_connect_provider_test.go:116]:
```suggestion
		PreCheck:     func() { testAccPreCheck(t) },
		ErrorCheck:   testAccErrorCheck(t, iam.EndpointsID),
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAWSInstanceProfileDestroy,
```

File: website/docs/r/iam_openid_connect_provider.html.markdown [hunk 3]
```diff
@@ -31,6 +31,7 @@ The following arguments are supported:
 * `url` - (Required) The URL of the identity provider. Corresponds to the _iss_ claim.
 * `client_id_list` - (Required) A list of client IDs (also known as audiences). When a mobile or web app registers with an OpenID Connect provider, they establish a value that identifies the application. (This is the value that's sent as the client_id parameter on OAuth requests.)
 * `thumbprint_list` - (Required) A list of server certificate thumbprints for the OpenID Connect (OIDC) identity provider's server certificate(s).
+* `tags` - Key-value mapping of tags for the IAM OIDC provider.
```
Comment (bflad) [inline website/docs/r/iam_openid_connect_provider.html.markdown:34]:
```suggestion
* `tags` - (Optional) Map of resource tags for the IAM OIDC provider.
```

**Review by bflad — APPROVED** [review]
Looks great, thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSIAMOpenIDConnectProvider_disappears (10.03s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_basic (20.50s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_tags (32.26s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSIAMOpenIDConnectProvider_disappears (15.58s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_basic (36.70s)
--- PASS: TestAccAWSIAMOpenIDConnectProvider_tags (47.27s)
```

**Pull request merged**